### ISSUE TITLE
V2 Brainstorming

### ISSUE THREAD

**Issue by weegeekps** [KhronosGroup/glTF-Project-Explorer#122]
This is an issue to brainstorm what V2 should be and to discuss what features we want.

**Comment by javagl** [comment]
Some topics are already addressed (or at least mentioned) on other issues:

- The project explorer should be applicable to other projects. 
  - This refers to projects that are "similar" to glTF (in that they specify standards or file formats), and where general resources (applications, tutorials, libraries) should be listed. One specific example is https://github.com/KhronosGroup/glTF-Project-Explorer/issues/85 
  - This _might_ also refer to "things that are not so similar to glTF", namely tutorials, videos, articles and presentations, as suggested in https://github.com/KhronosGroup/glTF-Project-Explorer/issues/88 
 It has to be decided where to draw the line - i.e. whether there are significant structural differences that require different structures to represent the two, or whether an "Entry" in such a project explorer instance can reasonably be generalized so that it can cover a WebApplication and a YouTube video, without introducing too many contortions on the implementation level...

- The search functionality should be improved (c.f. https://github.com/KhronosGroup/glTF-Project-Explorer/issues/119 ). 
  - For me, this is the biggest topic. I noticed that when I had been searching something, I did not use the search+filter at all, but instead, just used the CTRL+F browser search. That's not good. A large part of this could already be covered by not only taking the project _`title`_ into account, but also the _`description`_. An alternative would be to abandon the `task`, and instead, allow users to define `keywords` that are dedicatedly used as "search terms that should cause that project to show up". Some initial thoughts are in https://github.com/KhronosGroup/glTF-Project-Explorer/issues/111
  - One could consider using the search functionality of an existing CMS. The goal cannot be to re-implement some "on-site Google search"
  - One could consider some sort of auto-completion. For example, when the user types in `"C++` in the search field, it could offer some convenient, dropdown-like auto-completion with options like 
  
        [type:importer] 
        [type:exporter]
        [type:viewer]

    that can be used to "drill down" and refine the search, in an exploratory style

  - A "find similar projects" functionality would be nice. For example, someone might type in the name (or manually select) an existing "C++ glTF loader library with MIT license", and then hit a button to receive a list of other projects that match these tags. Yes, it will be necessary to quantify the _difference_ between "MIT" and "BSD", and between "C" and "C++" (ouch!), but it could be a _really_ useful feature for people who are looking for alternatives. 


- It might be interesting to have some sort of visualization of the ecosystem. 
   This was mentioned in https://github.com/KhronosGroup/glTF-Project-Explorer/issues/83 , but I wonder whether this will be an "integral part" of the explorer itself, or whether it could be a completely independent project that just pulls from the same database, or whether it will be a mix of both, namely a visualization that builds on top of some sort of search+filter API that the explorer offers (thus, keeping the explorer "standalone", and creating a "visualizer" that only depends on the explorer via a small, clean interface)

**Comment by weegeekps** [comment]
I don't have the user journeys completely typed up yet, but I do have the low-detail wire frames finished. There are 3 layouts presented here, one for the Basic Layout, the Detailed Info screen, and the Card Layout. For the Card Layout I've presented 3 variants to show how different repository links will be represented in the card list on the Basic Layout screen.

# Basic Layout
![Basic Layout](https://user-images.githubusercontent.com/284269/144645388-d9152c48-1760-4141-b19a-a0096eb53392.png)

# Detailed Info Layout
![Detailed Project Info Layout](https://user-images.githubusercontent.com/284269/144645447-97e38ceb-8d9d-4709-93ce-3926bdeaa3b3.png)

# Card Layout
![Card Layout](https://user-images.githubusercontent.com/284269/144645497-acca72d8-8872-4b3c-985d-2f82d7d31016.png)
![Card Layout (GitLab Icon)](https://user-images.githubusercontent.com/284269/144645531-de037363-0e53-456e-b3c8-8284d6c3e802.png)
![Card Layout (Link Icon)](https://user-images.githubusercontent.com/284269/144645542-6b557769-5712-4b2b-a805-4a42d5ad2b17.png)

**Comment by emackey** [comment]
It would be great if the projects that are established enough to have their own logos/icons could display them as you've shown in the final wireframe above.

**Comment by weegeekps** [comment]
I really like that idea, and I've spent the past few days mulling it over and seeing what we can do. I've narrowed down the possibilities for implementation to two options:

1. We allow users to upload a 64x64 SVG image of their logo, that will be referenced in the data file. This is my favorite option, but I believe we may run into potential legal issues if we don't require that the logo be licensed as CC-0 or CC-BY.
2. We allow users to link to a 64x64 SVG image that is on their site. I do not like this option, as it opens users up to several potential vulnerabilities.

I can't make the next Tooling TSG meeting but perhaps on the 17th we could discuss more about these options.

**Comment by emackey** [comment]
I don't think we can expect people's trademarks to be available as CC-anything.  But in cases where a trademark is used on a link to the legitimate website for the trademarked product, I believe it's often allowable (but standard disclaimer, I am not a lawyer).  This permission may need to be obtained on a product-by-product basis, however.

**Comment by weegeekps** [comment]
I am not a lawyer either, but I think the issue is the potential storage of the files. I do think you're right that we can't expect trademarks to be available as creative commons licensed. Probably worth a discussion with a lawyer.

**Comment by weegeekps** [comment]
I just opened #139 which replaces the simple title search with an indexed search. I'm going to leave it open for a week for comments.

Next up is redoing the design a bit. In order to make things a bit easier I am going to do a little refactoring to use something like tailwindcss. This will also help to solve a few issues with specific browsers. Should be done sometime next week with that PR.

**Comment by tim-lewis** [comment]
I really like the V2 layout, and the idea of including logos.  Has there been any thought on how the V2 layout would be responsive, especially in terms of the left 'filter' panel.  Would that work being pushed to the top of the page but with mobile friendly pull downs to limit height use, rather that a list of all the tags?  We are looking to do something similar to that with the right panel that appears on the khronos news sections at: https://www.khronos.org/news/

**Comment by javagl** [comment]
The filter panels are already moved to the top when the window width is below a certain threshold.

(I noticed a glitch here: The "Expand/Collapse" button is only available when they are at the top. When collapsing them, and then making the window wider, the collapsed panels move to the left, where they cannot be expanded any more... but that's a detail for now...)
